mslib/utils/config.py

```diff
@@ -0,0 +1,299 @@
+# -*- coding: utf-8 -*-
+"""
+
+    mslib.utils.config
+    ~~~~~~~~~~~~~~
+
+    Collection of utility routines for the Mission Support System.
+
+    This file is part of mss.
+
+    :copyright: Copyright 2008-2014 Deutsches Zentrum fuer Luft- und Raumfahrt e.V.
+    :copyright: Copyright 2011-2014 Marc Rautenhaus (mr)
+    :copyright: Copyright 2016-2021 by the mss team, see AUTHORS.
+    :license: APACHE-2.0, see LICENSE for details.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+"""
+
+import copy
+import json
+import logging
+import fs
+
+from mslib.utils import FatalUserError
+from mslib.msui import constants, MissionSupportSystemDefaultConfig
+from mslib.support.qt_json_view.datatypes import match_type, UrlType, StrType
+
+
+# system default options as dictionary
+default_options = dict(MissionSupportSystemDefaultConfig.__dict__)
```

Comment:
wouldn't it make sense to move the config class here? And maybe instrument it with the structure info stored in the dicts below?

Reply:
I'll look into this.